pgrx-pg-sys/src/submodules/ffi.rs

```diff
@@ -109,23 +112,27 @@ unsafe fn pg_guard_ffi_boundary_impl<T, F: FnOnce() -> T>(f: F) -> T {
         let caller_memxct = pg_sys::CurrentMemoryContext;
         let prev_exception_stack = pg_sys::PG_exception_stack;
         let prev_error_context_stack = pg_sys::error_context_stack;
-        let mut jump_buffer = std::mem::MaybeUninit::uninit();
-        let jump_value = crate::sigsetjmp(jump_buffer.as_mut_ptr(), 0);
-
-        if jump_value == 0 {
-            // first time through, not as the result of a longjmp
-            pg_sys::PG_exception_stack = jump_buffer.as_mut_ptr();
+        let mut result: std::mem::MaybeUninit<T> = MaybeUninit::uninit();
+        let jump_value = cee_scape::call_with_sigsetjmp(false, |jump_buffer| {
+            // Make Postgres' error-handling system aware of our new
+            // setjmp/longjmp restore point.
+            pg_sys::PG_exception_stack = std::mem::transmute(jump_buffer as *const SigJmpBufFields);
 
             // execute the closure, which will be a wrapped internal Postgres function
-            let result = f();
+            result.write(f());
+            0
+        });
 
-            // restore Postgres' understanding of where its next longjmp should go
+        if jump_value == 0 {
```

Comment:
I wonder if we'll need to tweak the scope this includes... maybe. This should be good for now, though.